apps/asap-cli/src/scripts/algolia/tests/process-productivity-metrics.test.ts

```diff
@@ -0,0 +1,252 @@
+import { SearchResponse } from '@algolia/client-search';
+import { teamOutputDocumentTypes, timeRanges } from '@asap-hub/model';
+import type { SearchIndex } from 'algoliasearch';
+
+import {
+  deletePreviousObjects,
+  getAllHits,
+  getBellCurveMetrics,
+  getStandardDeviation,
+  processTeamProductivityMetrics,
+  processUserProductivityMetrics,
+} from '../process-productivity-metrics';
+
+describe('getStandardDeviation', () => {
+  it('should return 0 for an empty array', () => {
+    expect(getStandardDeviation([], 0)).toBe(0);
+  });
+
+  it('should return correct standard deviation for an array of numbers', () => {
+    const numbers = [1, 2, 3, 4, 5];
+    const mean =
+      numbers.reduce((sum, value) => sum + value, 0) / numbers.length;
+    expect(getStandardDeviation(numbers, mean)).toBeCloseTo(1.414, 3);
+  });
+});
+
+describe('getBellCurveMetrics', () => {
+  it('should return correct bell curve metrics', () => {
+    const data = [1, 2, 3, 4, 5];
+    const metrics = getBellCurveMetrics(data);
+    expect(metrics).toEqual({
+      belowAverageMin: 1,
+      belowAverageMax: 1,
+      averageMin: 2,
+      averageMax: 4,
+      aboveAverageMin: 5,
+      aboveAverageMax: 5,
+    });
+  });
+});
+
+describe('getAllHits', () => {
+  it('should retrieve all hits across multiple pages', async () => {
+    const mockHits = Array.from({ length: 150 }, (_, i) => ({
+      objectID: `${i}`,
+    }));
+
+    const mockGetPaginatedHits = jest.fn((page) => {
+      const hitsPerPage = 50;
+      return Promise.resolve({
+        hits: mockHits.slice(page * hitsPerPage, (page + 1) * hitsPerPage),
+        nbPages: 3,
+      } as unknown as SearchResponse<(typeof mockHits)[0]>);
+    });
+
+    const allHits = await getAllHits(mockGetPaginatedHits);
+    expect(allHits).toHaveLength(150);
+    expect(mockGetPaginatedHits).toHaveBeenCalledTimes(3);
+  });
+});
+
+describe('deletePreviousObjects', () => {
+  it('should delete previous objects of specified type', async () => {
+    const mockIndex = {
+      search: jest.fn().mockResolvedValue({
+        hits: [{ objectID: '1' }, { objectID: '2' }],
+      }),
+      deleteObjects: jest.fn().mockResolvedValue({}),
+    } as unknown as SearchIndex;
+
+    await deletePreviousObjects(mockIndex, 'user-productivity');
+    expect(mockIndex.search).toHaveBeenCalledWith('', {
+      filters: '__meta.type:"user-productivity-metrics"',
+    });
+    expect(mockIndex.deleteObjects).toHaveBeenCalledWith(['1', '2']);
+  });
+});
+
+describe('processUserProductivityMetrics', () => {
+  it('should process user productivity metrics', async () => {
```

Comment:
what do you mean exactly, that it should save new records yes? I think you are also deleting before saving and this is not asserted so it's potentially open to regression